### PR TITLE
fix(clickhouse): support arrayMap and arrayFilter roundtrips [CODEX]

### PR DESCRIPTION
**Pull request by gaoflow**
## Problem

ClickHouse higher-order array functions `arrayMap(lambda, arr)` and `arrayFilter(lambda, arr)` were parsed as generic/unsupported function expressions instead of SQLGlot's semantic array function nodes.

This made ClickHouse -> ClickHouse roundtrips lossy for these functions, and it also meant `arrayMap` did not use the existing `exp.Transform` representation.

## Fix

- Add ClickHouse parser mappings for `ARRAYMAP` and `ARRAYFILTER`.
- Preserve ClickHouse's argument order by mapping `arrayMap(lambda, array)` to `exp.Transform(this=array, expression=lambda)` and `arrayFilter(lambda, array)` to `exp.ArrayFilter(this=array, expression=lambda)`.
- Add a ClickHouse generator transform for `exp.Transform` so `arrayMap` roundtrips back to ClickHouse syntax.

## Tests

Added focused ClickHouse identity tests:

- `arrayMap(x -> x + 1, arr)` roundtrips and parses as `exp.Transform`.
- `arrayFilter(x -> x > 0, arr)` roundtrips and parses as `exp.ArrayFilter`.

Local verification:

```text
python -m unittest tests.dialects.test_clickhouse.TestClickhouse.test_functions
python -m ruff check sqlglot/parsers/clickhouse.py sqlglot/generators/clickhouse.py tests/dialects/test_clickhouse.py
python -m ruff format --check sqlglot/parsers/clickhouse.py sqlglot/generators/clickhouse.py tests/dialects/test_clickhouse.py
```

---

This pull request was prepared with the assistance of AI, under my direction and review.

### CODE REVIEW

File: tests/dialects/test_clickhouse.py [hunk 1]
```diff
@@ -1768,6 +1768,28 @@ def test_functions(self):
...
+            read={
+                "duckdb": "SELECT LIST_TRANSFORM(arr, x -> x + 1) FROM t",
+                "spark": "SELECT TRANSFORM(arr, x -> x + 1) FROM t",
+            },
+            write={
+                "clickhouse": "SELECT arrayMap(x -> x + 1, arr) FROM t",
+                "duckdb": "SELECT LIST_TRANSFORM(arr, x -> x + 1) FROM t",
+                "spark": "SELECT TRANSFORM(arr, x -> x + 1) FROM t",
+            },
+        )
```
Comment (geooo109) [inline tests/dialects/test_clickhouse.py]: There are cases that the transpilation results into different semantics.

```
clickhouse (input):
SELECT arrayMap(NULL, [1, 2]) AS res

Query id: 0db4a81e-03b1-4253-8593-6a73112a447f

   ┌─res──┐
1. │ ᴺᵁᴸᴸ │
   └──────┘

1 row in set. Elapsed: 0.002 sec. 

transpiled duckdb (output):
memory D SELECT LIST_TRANSFORM([1,2], NULL) AS res;
Binder Error:
Invalid lambda expression!

transpiled spark 4 (output):
spark-sql (default)> SELECT TRANSFORM(array(1,2), NULL);
[null,null]
Time taken: 0.051 seconds, Fetched 1 row(s)
```

File: tests/dialects/test_clickhouse.py [hunk 2]
```diff
@@ -1768,6 +1768,28 @@ def test_functions(self):
...
+        self.validate_all(
+            "SELECT arrayFilter(x -> x > 0, arr) FROM t",
+            read={
+                "duckdb": "SELECT LIST_FILTER(arr, x -> x > 0) FROM t",
+            },
+            write={
+                "clickhouse": "SELECT arrayFilter(x -> x > 0, arr) FROM t",
+                "duckdb": "SELECT LIST_FILTER(arr, x -> x > 0) FROM t",
+            },
+        )
```
Comment (geooo109) [inline tests/dialects/test_clickhouse.py]: Same as https://github.com/tobymao/sqlglot/pull/7794/changes#r3473795554 :
```
clickhouse (input):
SELECT arrayFilter(NULL, [1, 2, 3])

Query id: 2b70a67e-8db5-47e5-b60a-02551e878843

   ┌─arrayFilter(NULL, [1, 2, 3])─┐
1. │ ᴺᵁᴸᴸ                         │
   └──────────────────────────────┘

1 row in set. Elapsed: 0.002 sec. 

transpiled duckdb(output): 
memory D SELECT LIST_FILTER([1, 2, 3], NULL);
Binder Error:
Invalid lambda expression!
```

File: sqlglot/parsers/clickhouse.py [hunk 3]
```diff
@@ -262,6 +262,14 @@ class ClickHouseParser(parser.Parser):
         "ARRAYMIN": exp.ArrayMin.from_arg_list,
         "ARRAYREVERSE": exp.ArrayReverse.from_arg_list,
         "ARRAYSLICE": exp.ArraySlice.from_arg_list,
+        # ClickHouse higher-order array functions: the lambda comes first, the array second.
+        # This is the opposite of exp.ArrayFilter(this=array, expression=lambda) convention.
```
Comment (geooo109) [inline sqlglot/parsers/clickhouse.py]: Let's remove the comments here, to simplify the code.

**Review by geooo109 — COMMENTED** [review]
Nice work, I left some comments. 

My suggestion is the following:

Remove the current `validate_all` tests of this PR, and only use `vaildate_identity` tests for testing the roundtrip (input clickhouse -> output clickhouse) of the added functions for example `self.validate_identity("arrayFilter(x -> x > 0, arr)").assert_is(exp.ArrayFilter)`. This would make the PR simple and mergable.

Then, if you want you can investigate various inputs - outputs for the transpilation, as you saw the `NULL` creates semantic issues for the transpilation. After, detecting all this edge cases let's make a following PR that contains the robust transpilation.

**Comment by binggao1230** [comment]
Addressed in `806c7698`:

- removed the `validate_all` cross-dialect transpilation tests
- kept only focused ClickHouse `validate_identity` coverage for `arrayMap` / `arrayFilter`, with AST type assertions
- removed the parser comments to keep the code smaller

I also updated the PR title and description to reflect the narrowed roundtrip scope.

**Review by geooo109 — APPROVED** [review]
thank you, great work!

**Review by georgesittas — APPROVED** [review]
